src/android/FirebasePlugin.java

```diff
@@ -1912,6 +1912,50 @@ public void run() {
         });
     }
 
+    public void clearNotification(final CallbackContext callbackContext, final JSONArray options) {
+        cordova.getThreadPool().execute(new Runnable() {
+            public void run() {
+                try {
+                    NotificationManager nm = (NotificationManager) applicationContext.getSystemService(Context.NOTIFICATION_SERVICE);
+                    String id = options.getString(0);
+                    if (options.isNull(1)) {
+                        nm.cancel(id.hashCode());
+                    } else {
+                        String tag = options.getString(1);
+                        nm.cancel(tag, id.hashCode());
+                    }
+                    callbackContext.success();
+                } catch (Exception e) {
+                    handleExceptionWithContext(e, callbackContext);
+                }
+            }
+        });
+    }
+
+    public void getAllNotifications(final CallbackContext callbackContext) {
+        cordova.getThreadPool().execute(new Runnable() {
+            public void run() {
+                // only call on Android M and above
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
Does this mean on onlder Androids we'll never clear some notifications (leaving them to be manually cleared only)?

Reply:
Yeah, although really old versions (v6), so serves them right